### ISSUE TITLE
Uncatchable `RuntimeError` with `No type found for...`

### ISSUE THREAD

**Issue by keithpitt** [rmosolgo/graphql-ruby#626]
So I stumbled across this GraphQL query today:

```graphql
query Index_ViewerRelayQL($msg_0: msg!) {
  node(id: $id_0) {
    id
    changelogs(read: false) @include(if: true) {
      count
    }
  }
}
```

`$msg_0` is totally invalid in this case. When I run it, I get un un-rescuable error:

```
RuntimeError: No type found for 'msg'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/schema/type_map.rb:19:in `[]'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/schema/type_expression.rb:9:in `build_type'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/schema/type_expression.rb:12:in `build_type'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/schema.rb:226:in `type_from_ast'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query/variables.rb:21:in `block in initialize'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query/variables.rb:16:in `each'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query/variables.rb:16:in `each_with_object'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query/variables.rb:16:in `initialize'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query.rb:149:in `new'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query.rb:149:in `variables'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query.rb:202:in `arguments_for'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/execution/directive_checks.rb:25:in `block in include?'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/execution/directive_checks.rb:15:in `each'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/execution/directive_checks.rb:15:in `include?'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/internal_representation/rewrite.rb:202:in `skip?'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/internal_representation/rewrite.rb:85:in `block in validate'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:65:in `block in apply_hooks'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:65:in `each'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:65:in `reduce'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:65:in `apply_hooks'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:54:in `begin_visit'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:45:in `visit_node'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `block in visit_node'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `each'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `visit_node'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `block in visit_node'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `each'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `visit_node'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `block in visit_node'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `each'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:47:in `visit_node'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/language/visitor.rb:39:in `visit'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/static_validation/validator.rb:34:in `validate'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query.rb:239:in `document_valid?'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query.rb:214:in `valid?'
/Users/keithpitt/.gem/ruby/2.4.0/gems/graphql-1.5.3/lib/graphql/query.rb:119:in `result'
```

Which then bubbles up to my error tracker. Interestingly...if I change it to (removing the `if`)

```graphql
query Index_ViewerRelayQL($msg_0: msg!) {
  node(id: $id_0) {
    id
    changelogs(read: false)  {
      count
    }
  }
}
```

GraphQL seems happier about it, and it returns a proper error:

```json
{
  "errors": [
    {
      "message": "msg isn't a defined input type (on $msg_0)",
      "locations": [
        {
          "line": 1,
          "column": 27
        }
      ],
      "fields": [
        "query Index_ViewerRelayQL"
      ]
    },
    {
      "message": "Field 'changelogs' doesn't exist on type 'Node'",
      "locations": [
        {
          "line": 4,
          "column": 5
        }
      ],
      "fields": [
        "query Index_ViewerRelayQL",
        "node",
        "changelogs"
      ]
    },
    {
      "message": "Variable $msg_0 is declared by Index_ViewerRelayQL but not used",
      "locations": [
        {
          "line": 1,
          "column": 1
        }
      ],
      "fields": [
        "query Index_ViewerRelayQL"
      ]
    },
    {
      "message": "Variable $id_0 is used by Index_ViewerRelayQL but not declared",
      "locations": [
        {
          "line": 2,
          "column": 12
        }
      ],
      "fields": [
        "query Index_ViewerRelayQL",
        "node",
        "id"
      ]
    }
  ]
}
```

**Comment by rmosolgo** [comment]
Oh my! I wonder why the one error sneaks through but the other one is handled properly. I mean to fix this in #603 but apparently I missed a spot. Thanks for the great bug report, I'll dig in in the morning!

**Comment by keithpitt** [comment]
All good, thanks for being awesome!
